addons/im_livechat/static/src/core/common/thread_actions.js

```diff
@@ -0,0 +1,28 @@
+import { threadActionsRegistry } from "@mail/core/common/thread_actions";
+
+const actinConfig = threadActionsRegistry.get("view-contact");
+
+threadActionsRegistry.add("livechat-visitorPartner-contact", {
+    ...actinConfig,
+    condition(component) {
+        return (
+            component.thread?.channel_type === "livechat" &&
+            component.thread?.livechatVisitorMember?.persona.type === "partner"
+        );
+    },
+    open(component) {
+        const visitor = component.thread.livechatVisitorMember;
+        component.actionService?.doAction(
+            {
+                type: "ir.actions.act_window",
+                res_model: "res.partner",
+                views: [[false, "form"]],
+                search_view_id: [false],
+                res_id: visitor?.persona.id,
+            },
+            { newWindow: !!component.env.inDiscussApp }
```

Comment:
Opening in new page doesn't feel great. It should always open in current window.
If discuss app is open, the current channel can be open as chat window at the same time as open the form view.